Unable to create table column for property [id] of entity with unknown data type: OBJECT for Groovy and Maven

Projects created with Starter for Groovy and Maven doesn't work with Data-JDBC. Not 100% sure if there is an error here or a missconfiguration in the generated project.

### Steps to Reproduce

- Create a new application with Groovy/Maven/Spock and `data-jdbc`. Download it directly [here](https://micronaut.io/launch/?javaVersion=JDK_11&lang=GROOVY&build=MAVEN&test=SPOCK&name=demo&package=com.example&type=DEFAULT&features=data-jdbc&version=2.3.4&activity=create).
- Add the following entity class:

```groovy
package com.example

import io.micronaut.data.annotation.GeneratedValue
import io.micronaut.data.annotation.Id
import io.micronaut.data.annotation.MappedEntity

@MappedEntity
class RefreshTokenEntity {
    @Id
    @GeneratedValue
    Long id
}
```
- Run `./mvnw verify`

### Expected Behaviour

The test should pass

### Actual Behaviour

The test fails with:

```
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running DemoSpec
11:51:27.012 [main] INFO  i.m.context.env.DefaultEnvironment - Established active environments: [test]
11:51:27.391 [main] INFO  com.zaxxer.hikari.HikariDataSource - HikariPool-1 - Starting...
11:51:27.532 [main] INFO  com.zaxxer.hikari.HikariDataSource - HikariPool-1 - Start completed.
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.764 s <<< FAILURE! - in DemoSpec
[ERROR] DemoSpec  Time elapsed: 0.763 s  <<< ERROR!
io.micronaut.context.exceptions.BeanInstantiationException: Bean definition [io.micronaut.data.jdbc.config.SchemaGenerator] could not be loaded: Error instantiating bean of type [io.micronaut.data.jdbc.config.SchemaGenerator]: Unable to create table column for property [id] of entity [com.example.RefreshTokenEntity] with unknown data type: OBJECT
Caused by: io.micronaut.context.exceptions.BeanInstantiationException: Error instantiating bean of type [io.micronaut.data.jdbc.config.SchemaGenerator]: Unable to create table column for property [id] of entity [com.example.RefreshTokenEntity] with unknown data type: OBJECT
Caused by: io.micronaut.data.exceptions.MappingException: Unable to create table column for property [id] of entity [com.example.RefreshTokenEntity] with unknown data type: OBJECT

[INFO] 
[INFO] Results:
[INFO] 
[ERROR] Errors: 
[ERROR]   DemoSpec » BeanInstantiation Bean definition [io.micronaut.data.jdbc.config.Sc...
[INFO] 
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
```

### Environment Information

- **Operating System**: Linux Mint 20.1
- **Micronaut Version:** 2.3.4
- **JDK Version:** 8 & 11

--- a/starter-core/src/main/java/io/micronaut/starter/feature/build/maven/templates/pom.rocker.raw
+++ b/starter-core/src/main/java/io/micronaut/starter/feature/build/maven/templates/pom.rocker.raw
@@ -106,6 +106,9 @@ MavenBuild mavenBuild
 @if (features.language().isGroovy()) {
 @dependency.template("io.micronaut", "micronaut-inject-groovy", "provided", null)
 @dependency.template("org.codehaus.groovy", "groovy", "compile", "${groovyVersion}")
+@for (processor : mavenBuild.getAnnotationProcessors()) {
+@dependency.template(processor.getGroupId(), processor.getArtifactId(), "provided", null)
+}
 }
 @dependency.template("io.micronaut", "micronaut-inject", "compile", null)
 @dependency.template("io.micronaut", "micronaut-validation", "compile", null)

--- a/starter-core/src/test/groovy/io/micronaut/starter/feature/database/DataJdbcSpec.groovy
+++ b/starter-core/src/test/groovy/io/micronaut/starter/feature/database/DataJdbcSpec.groovy
@@ -8,6 +8,7 @@ import io.micronaut.starter.feature.Features
 import io.micronaut.starter.fixture.CommandOutputFixture
 import io.micronaut.starter.options.BuildTool
 import io.micronaut.starter.options.Language
+import spock.lang.Issue
 
 class DataJdbcSpec extends ApplicationContextSpec  implements CommandOutputFixture {
 
@@ -132,5 +133,23 @@ class DataJdbcSpec extends ApplicationContextSpec  implements CommandOutputFixtu
         then:
         ctx.configuration.containsKey("datasources.default.url")
         ctx.configuration.get("datasources.default.dialect") == "SQL_SERVER"
+    }
+
+    @Issue("https://github.com/micronaut-projects/micronaut-starter/issues/686")
+    void 'test data-processor dependency is in provided scope for Groovy and Maven'() {
+        when:
+        String template = new BuildBuilder(beanContext, BuildTool.MAVEN)
+            .language(Language.GROOVY)
+            .features(["data-jdbc"])
+            .render()
+
+        then:
+        template.contains('''
+    <dependency>
+      <groupId>io.micronaut.data</groupId>
+      <artifactId>micronaut-data-processor</artifactId>
+      <scope>provided</scope>
+    </dependency>
+''')
     }
 }

--- a/starter-core/src/test/groovy/io/micronaut/starter/feature/database/DataJpaSpec.groovy
+++ b/starter-core/src/test/groovy/io/micronaut/starter/feature/database/DataJpaSpec.groovy
@@ -7,6 +7,7 @@ import io.micronaut.starter.application.generator.GeneratorContext
 import io.micronaut.starter.feature.Features
 import io.micronaut.starter.options.BuildTool
 import io.micronaut.starter.options.Language
+import spock.lang.Issue
 
 class DataJpaSpec extends ApplicationContextSpec {
 
@@ -150,5 +151,23 @@ class DataJpaSpec extends ApplicationContextSpec {
         then:
         ctx.configuration.containsKey("datasources.default.url")
         ctx.configuration.get("datasources.default.dialect") == "SQL_SERVER"
+    }
+
+    @Issue("https://github.com/micronaut-projects/micronaut-starter/issues/686")
+    void 'test data-processor dependency is in provided scope for Groovy and Maven'() {
+        when:
+        String template = new BuildBuilder(beanContext, BuildTool.MAVEN)
+            .language(Language.GROOVY)
+            .features(["data-jpa"])
+            .render()
+
+        then:
+        template.contains('''
+    <dependency>
+      <groupId>io.micronaut.data</groupId>
+      <artifactId>micronaut-data-processor</artifactId>
+      <scope>provided</scope>
+    </dependency>
+''')
     }
 }

